File: CSC454HW3/CSC454HW3/src/csc454hw3/XOR.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package csc454hw3;

/**
 *
 * @author chrisrk192
 */
public class XOR implements ModelInterface{
    private int state;
    
    
    
    @Override
    public String GetInitilizationSignature() {
        return "initialState";
    }

    @Override
    public void Initialize(String[] args) {
        state = Integer.parseInt(args[0]);
    }

    @Override
    public String[] OutPut() {
        String[] s = new String[1];
        s[0] = state + "";
        return s;
    }

    @Override
    public void StateTransition(String[] inputItems) throws CantPerformAction {
        int one, two;
        if(inputItems.length < 2)
            throw new CantPerformAction("not enough inputs");
        try{
            one = Integer.parseInt(inputItems[0]);
        } catch(NumberFormatException e){
            throw new CantPerformAction("Input invalid: \""+inputItems[0]+"\"");
        }
        try{
            two = Integer.parseInt(inputItems[1]);
        } catch(NumberFormatException e){
            throw new CantPerformAction("Input invalid: \""+inputItems[1]+"\"");
        }
        state =  one ^ two;
    }

    @Override
    public String GetStatus() {
        return "state = "+state;
    }

    @Override
    public Object Clone() {
        XOR x = new XOR();
        x.state = this.state;
        return x;
    }

    @Override
    public String[] Tick(String[] args) throws CantPerformAction{
        String[] out = OutPut();
        StateTransition(args);
        return out;
    }
    
}

File: CSC454HW5/CppImpOfHW4/XOR.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   XORModel.cpp
 * Author: chrisrk192
 * 
 * Created on October 27, 2017, 10:21 PM
 */

#include "XOR.h"

XOR::XOR() {
}

XOR::XOR(const XOR& orig) {
}

XOR::~XOR() {
}

std::string XOR::GetInitilizationSignature(){
    return "initial state";
}
void XOR::Initialize(std::vector<std::string> inputs ){
    state = std::stoi(inputs.at(0));
}
std::vector<std::string> XOR::OutPut(){
    std::vector<std::string> o;
    o.push_back(std::to_string(state));
    return o;
}  
void XOR::StateTransition(std::vector<std::string> inputItems){
    int one, two;
        if(inputItems.size() < 2)
            throw new CantPerformAction("not enough inputs");
            one = stoi(inputItems.at(0));
            two = stoi(inputItems.at(1));
        state =  one ^ two;
}
std::string XOR::GetStatus(){
    return "state = "+std::to_string(state);
}
std::vector<std::string> XOR::Tick(std::vector<std::string> inputs){
    std::vector<std::string> out = OutPut();
    StateTransition(inputs);
    return out;
}